api/src/shared/swaggers.js

```diff
@@ -4,88 +4,98 @@ import packageJSON from '../../package.json' with { type: 'json' };
 import { config } from './config.js';
 import { logger } from './infrastructure/utils/logger.js';
 
-const swaggerOptionsAuthorizationServer = {
-  routeTag: 'authorization-server',
-  info: {
-    title: 'Welcome to the Pix Authorization server',
-    version: packageJSON.version,
-  },
-  jsonPath: '/swagger.json',
-};
+class PixOpenApiBaseDefinition {
+  constructor() {
+    this.swaggerConfiguration = {
+      OAS: 'v3.0',
+      basePath: '/api',
+      routeTag: 'api',
+      info: {
+        version: packageJSON.version,
+      },
+      jsonPath: '/swagger.json',
+      documentationPath: '/documentation',
+      swaggerUIPath: '/documentation/swaggerui/',
+      uiOptions: {
+        url: 'swagger.json',
+      },
+      securityDefinitions: {
+        bearerAuth: {
+          name: 'Authorization',
+          scheme: 'Bearer',
+          in: 'header',
+          description: 'Example: Bearer eyJ...z',
+          type: 'apiKey',
+        },
+      },
+      security: [{ bearerAuth: [], jwt: [] }],
+    };
+  }
+}
 
-const swaggerOptionsLivretScolaire = {
-  routeTag: 'livret-scolaire',
-  info: {
-    title: 'Welcome to the Pix LSU/LSL open api',
-    version: packageJSON.version,
-  },
-  jsonPath: '/swagger.json',
-};
+class AuthorizationServer extends PixOpenApiBaseDefinition {
+  constructor() {
+    super();
+    this.swaggerConfiguration.info.title = 'Welcome to the Pix Authorization server';
+    this.swaggerConfiguration.routeTag = 'authorization-server';
+  }
+}
 
-const swaggerOptionsPoleEmploi = {
-  routeTag: 'pole-emploi',
-  info: {
-    title: 'Pix Pôle emploi open api',
-    version: packageJSON.version,
-  },
-  jsonPath: '/swagger.json',
-};
+class LivretScolaire extends PixOpenApiBaseDefinition {
+  constructor() {
+    super();
+    this.swaggerConfiguration.info.title = 'Welcome to the Pix LSU/LSL open api';
+    this.swaggerConfiguration.routeTag = 'livret-scolaire';
+  }
+}
 
-const swaggerOptionsParcoursup = {
-  routeTag: 'parcoursup',
-  OAS: 'v3.0',
-  servers: _buildParcoursupServers(),
-  pathReplacements: [
-    {
-      replaceIn: 'all',
-      pattern: /api\/application\//,
-      replacement: '',
-    },
-  ],
-  info: {
-    title: 'Pix Parcoursup Open Api',
-    version: packageJSON.version,
-  },
-  jsonPath: '/swagger.json',
-  securityDefinitions: {
-    bearerAuth: {
-      name: 'Authorization',
-      scheme: 'Bearer',
-      in: 'header',
-      description: 'Example: Bearer eyJ...z',
-      type: 'apiKey',
-    },
-  },
-  security: [{ bearerAuth: [], jwt: [] }],
-};
+class PoleEmploi extends PixOpenApiBaseDefinition {
+  constructor() {
+    super();
+    this.swaggerConfiguration.info.title = 'Pix Pôle emploi open api';
```

Comment:
sinon pour être iso à Parcoursup

```suggestion
    this.swaggerConfiguration.info.title = 'Pix Pôle emploi Open Api';
```

Reply:
Done

Reply:
Done !